anvil/src/eth/backend/mem/mod.rs

```diff
@@ -630,14 +637,25 @@ impl Backend {
     /// Write all chain data to serialized bytes buffer
     pub async fn dump_state(&self) -> Result<Bytes, BlockchainError> {
         let state = self.serialized_state().await?;
-        let content = serde_json::to_vec(&state).unwrap_or_default().into();
-        Ok(content)
+        let mut e = GzEncoder::new(Vec::new(), Compression::default());
+        e.write_all(&serde_json::to_vec(&state).unwrap_or_default())
+            .map_err(|_| BlockchainError::DataUnavailable)?;
+        Ok(e.finish().unwrap_or_default().into())
     }
 
     /// Deserialize and add all chain data to the backend storage
     pub async fn load_state(&self, buf: Bytes) -> Result<bool, BlockchainError> {
-        let state: SerializableState =
-            serde_json::from_slice(&buf.0).map_err(|_| BlockchainError::FailedToDecodeStateDump)?;
+        let orig_buf = &buf.0[..];
+        let mut d = GzDecoder::new(orig_buf);
+        let mut v = Vec::new();
+
+        let state: SerializableState = serde_json::from_slice(if d.header().is_some() {
+            d.read_to_end(v.as_mut()).map_err(|_| BlockchainError::FailedToDecodeStateDump)?;
```

Comment:
hmm does the `map_err` return an err? if so we could maybe have a variant that includes the error

Reply:
the `read_to_end` emits an io::Result. If there is a problem, the most description that could probably be returned to the user is probably `Invalid Format` or `Unexpected Token` or `Invalid Length` etc, things that probably wouldn't be more helpful than just returning a generic decoder error. I could add a distsinction between decoding failure of the serde_json vs. decompressing of data, but again what difference in action would that actually lead the user? They aren't supposed to be modifying state dumps by hand.

Reply:
Yeah makes sense actually. It's alright then—this wouldn't surface extra info